src/main/java/org/aion/wallet/connector/api/ApiBlockchainConnector.java

```diff
@@ -275,6 +294,75 @@ private void handleSettingsChanged(final SettingsEvent event) {
         }
     }
 
+    private void processTransactionsOnReconnect() {
+        final Set<String> addresses = getAccountManager().getAddresses();
+        final BlockDTO oldestSafeBlock = getOldestSafeBlock(addresses, i -> {
+        });
+        processTransactionsFromBlock(oldestSafeBlock, addresses);
+    }
+
+    private void processTransactionsFromOldestRegisteredSafeBlock() {
+        final Set<String> addresses = getAccountManager().getAddresses();
+        final Consumer<Iterator<String>> nullSafeBlockFilter = Iterator::remove;
+        final BlockDTO oldestSafeBlock = getOldestSafeBlock(addresses, nullSafeBlockFilter);
+        if (oldestSafeBlock != null) {
+            processTransactionsFromBlock(oldestSafeBlock, addresses);
+        }
+    }
+
+    private BlockDTO getOldestSafeBlock(final Set<String> addresses, final Consumer<Iterator<String>> nullSafeBlockFilter) {
+        BlockDTO oldestSafeBlock = null;
+        final Iterator<String> addressIterator = addresses.iterator();
+        while (addressIterator.hasNext()) {
+            final String address = addressIterator.next();
+            final BlockDTO lastSafeBlock = getAccountManager().getLastSafeBlock(address);
+            if (lastSafeBlock != null) {
+                if (oldestSafeBlock == null) {
```

Comment:
can be refactored into `oldestSafeBlock == null || oldestSafeBlock.getNumber() > lastSafeBlock.getNumber()`

Reply:
will do